src/db/models.py

```diff
@@ -1,48 +1,101 @@
 from . import connection
 import psycopg2.sql as sql
+import abc
 
-class ApplicationsModel:
-    TableName = "applications"
+class BaseModel(abc.ABC):
+    TableName = ""
     Model = {}
 
     @classmethod
-    def setModel(cls, fieldnames, sample_data):
-        for field in fieldnames:
-            cls.Model[field] = str # force everything into string, deal with on client
-        
-        cls.Model["score"] = int
-        
+    def dropTable(cls):
+        conn = connection.get_connection()
+        cursor = conn.cursor()
+        try:
+            cursor.execute('DROP TABLE IF EXISTS "%s"' % cls.TableName)
+            conn.commit()
+        except Exception as e:
+            raise(e)
+        finally:
+            cursor.close()
+            conn.close()
+
+    @classmethod
+    def createTable(cls):
         conn = connection.get_connection()
         cursor = conn.cursor()
         try:
-            cursor.execute('DROP TABLE IF EXISTS %s' % ApplicationsModel.TableName)
-            cursor.execute('CREATE TABLE %s ()' % ApplicationsModel.TableName)
+            cursor.execute('CREATE TABLE "%s" ()' % cls.TableName)
             
             for name, data_type in cls.Model.items():
+                if type(data_type) is str:
```

Comment:
Is there a need to allow the caller to set postgre_type directly?

Reply:
yea it should only contain python types